node.gyp

```diff
@@ -290,7 +290,7 @@
           '-Wl,-bnoerrmsg',
         ],
       }],
-      ['(OS=="linux" or OS=="mac") and llvm_version!=0', {
+      ['(OS=="linux" or OS=="mac") and llvm_version!="0.0"', {
```

Comment:
Failure on macOS seems to be because of this change but I don't really understand why. The condition should have been true before as well.

Reply:
> Do .gyp files support the syntax:
> `if OS in ("linux", "mac")` ?

Yes, I pushed a commit to change that.

Reply:
The build is failing because `llvm_version` is still `0` (returned from `get_version_helper()` which hasn't been changed).

I'm not sure about the change to `"0.0"` though because I don't think version comparisons using strings is safe (e.g. "2.3" is considered > "2.26").

Reply:
> The build is failing because `llvm_version` is still `0` (returned from `get_version_helper()` which hasn't been changed).
> 
> I'm not sure about the change to `"0.0"` though because I don't think version comparisons using strings is safe (e.g. "2.3" is considered > "2.26").

Does it mean that the checks have always been broken?

Reply:
@richardlau I'm pretty sure the version number checks have always been subject to this issue. This is relatively serious but unrelated to my change, which just fixes the type inconsistency (Python 3 raises an error if you try to compare an int with a string).

Reply:
Is https://github.com/nodejs/node-gyp/issues/1917 related?